dss-apps/dss-datamodel-center/dss-datamodel-table-management/src/main/java/com/webank/wedatasphere/dss/datamodel/table/service/impl/LabelServiceImpl.java

```diff
@@ -0,0 +1,203 @@
+import com.webank.wedatasphere.dss.datamodel.table.dto.LabelQueryDTO;
+import com.webank.wedatasphere.dss.datamodel.table.entity.DssDatamodelLabel;
+import com.webank.wedatasphere.dss.datamodel.table.entity.DssDatamodelTable;
+import com.webank.wedatasphere.dss.datamodel.table.service.LabelService;
+import com.webank.wedatasphere.dss.datamodel.table.vo.LabelAddVO;
+import com.webank.wedatasphere.dss.datamodel.table.vo.LabelEnableVO;
+import com.webank.wedatasphere.dss.datamodel.table.vo.LabelUpdateVO;
+import com.webank.wedatasphere.dss.datamodel.table.vo.LabelsQueryVO;
+import org.apache.linkis.common.exception.ErrorException;
+import org.apache.linkis.server.Message;
+import org.apache.commons.lang3.StringUtils;
+import org.modelmapper.ModelMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+@Service
+public class LabelServiceImpl extends ServiceImpl<DssDatamodelLabelMapper, DssDatamodelLabel> implements LabelService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LabelServiceImpl.class);
+
+    private final Gson gson = new Gson();
+
+    @Resource
+    private DatamodelReferencService datamodelReferencService;
+
+    @Resource
+    private ModelMapper modelMapper;
+
+    @Resource
+    private AssertsSyncService assertsSyncService;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public Long add(LabelAddVO vo) throws ErrorException {
+        if (getBaseMapper().selectCount(Wrappers.<DssDatamodelLabel>lambdaQuery().eq(DssDatamodelLabel::getName, vo.getName())
+                .or().eq(DssDatamodelLabel::getFieldIdentifier,vo.getFieldIdentifier())) > 0) {
+            LOGGER.error("errorCode : {}, label name or field identifier can not repeat, name : {}", ErrorCode.LABEL_ADD_ERROR.getCode(), vo.getName());
+            throw new DSSDatamodelCenterException(ErrorCode.LABEL_ADD_ERROR.getCode(), "label name or field identifier can not repeat");
+        }
+
+        if (datamodelReferencService.labelReferenceCount(vo.getName())>0
+                ||datamodelReferencService.labelReferenceCount(vo.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, label name can not be referenced", ErrorCode.DIMENSION_ADD_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.DIMENSION_ADD_ERROR.getCode(), "label name can not be referenced");
+        }
+
+
+        DssDatamodelLabel newOne = modelMapper.map(vo,DssDatamodelLabel.class);
+        newOne.setParams(gson.toJson(vo.getParamMap()));
+        newOne.setCreateTime(new Date());
+        newOne.setUpdateTime(new Date());
+
+        getBaseMapper().insert(newOne);
+
+        //todo 同步资产创建标签
+        assertsSyncService.syncCreateLabel(new CreateLabelEvent(this
```

Comment:
publisher.publishEvent(new CreateLabelEvent(this
                , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                , vo.getName()));